File: src/main/java/org/intellij/plugin/mdx/lang/psi/MdxFile.java
package org.intellij.plugin.mdx.lang.psi;

import com.intellij.extapi.psi.PsiFileBase;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.psi.FileViewProvider;
import org.intellij.plugin.mdx.lang.MdxFileType;
import org.intellij.plugin.mdx.lang.MdxLanguage;
import org.jetbrains.annotations.NotNull;

public class MdxFile extends PsiFileBase {
    protected MdxFile(@NotNull FileViewProvider viewProvider) {
        super(viewProvider, MdxLanguage.INSTANCE);
    }

    @Override
    public @NotNull FileType getFileType() {
        return MdxFileType.INSTANCE;
    }

    @Override
    public String toString() {
        return "MDX File";
    }
}

File: src/main/java/org/intellij/plugin/mdx/lang/psi/MdxFile.kt
package org.intellij.plugin.mdx.lang.psi

import com.intellij.extapi.psi.PsiFileBase
import com.intellij.lang.javascript.psi.JSFile
import com.intellij.lang.javascript.psi.JSSourceElement
import com.intellij.lang.javascript.psi.impl.JSFileBaseImpl
import com.intellij.lang.javascript.psi.impl.JSFileImpl
import com.intellij.openapi.fileTypes.FileType
import com.intellij.psi.FileViewProvider
import org.intellij.plugin.mdx.lang.MdxFileType
import org.intellij.plugin.mdx.lang.MdxLanguage
//PsiFileBase(viewProvider!!, MdxLanguage),
class MdxFile(viewProvider: FileViewProvider?) : PsiFileBase(viewProvider!!, MdxLanguage) {
    override fun getFileType(): FileType {
        return MdxFileType.INSTANCE
    }

    override fun toString(): String {
        return "MDX File"
    }
}